x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_drilldowns_services.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup } from 'src/core/public';
+import { SetupDependencies } from '../../plugin';
+import {
+  CONTEXT_MENU_TRIGGER,
+  EmbeddableContext,
+} from '../../../../../../src/plugins/embeddable/public';
+import {
+  FlyoutCreateDrilldownAction,
+  FlyoutEditDrilldownAction,
+  OPEN_FLYOUT_ADD_DRILLDOWN,
+  OPEN_FLYOUT_EDIT_DRILLDOWN,
+} from './actions';
+import { DrilldownsStartContract } from '../../../../drilldowns/public';
+import { DashboardToDashboardDrilldown } from './dashboard_to_dashboard_drilldown';
+
+declare module '../../../../../../src/plugins/ui_actions/public' {
```

Comment:
this is generic - any application with embeddables could use these flyouts to create drilldowns (though they will need to handle storage/persistence).  Move to `drilldowns` or `advance_ui_Actions`?

Reply:
We can move these files there. I've added this to our sync today to discuss.

Things to consider:

- They are registered under `xpack.dashboardEnhanced.drilldowns.enabled` config setting, would be nice to keep at least the registration in dashboard for now, otherwise I will need to redo the feature flag. Also, if it is a feature of dashboard, maybe it should stay in dashboard? At least the registration part.
- Currently we don't have any other use cases. And all other places that have drilldowns&mdash;APM, Graph, ML&mdash;don't use embeddables.
- Looks like they are generic, at least mostly. Maybe embeddables can have different inputs.